sdk/personalizer/Azure.AI.Personalizer/src/Models/RlObjectConverter.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using Rl.Net;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.AI.Personalizer
+{
+    /// <summary> The converter between objects for Rl.Net objects and the sdk </summary>
+    internal static class RlObjectConverter
+    {
+        /// <summary>
+        /// Convert PersonalizerRankOptions object to a json context string for Rl.Net
+        /// </summary>
+        public static string ConvertToContextJson(PersonalizerRankOptions personalizerRankOptions)
+        {
+            DecisionContext decisionContext = new DecisionContext(personalizerRankOptions);
+            return JsonSerializer.Serialize(decisionContext);
+        }
+
+        /// <summary>
+        /// Create rank result based on Rl.Net response
+        /// </summary>
+        public static PersonalizerRankResult GenerateRankResult(List<PersonalizerRankableAction> originalActions,
+     List<PersonalizerRankableAction> rankableActions, List<PersonalizerRankableAction> excludedActions, RankingResponse rankingResponse, string eventId)
+        {
+            var rankedIndices = rankingResponse?.Select(actionProbability => ((int)actionProbability.ActionIndex + 1)).ToArray();
+
+            var rankingProbabilities = rankingResponse?.Select(actionProbability =>
+                actionProbability.Probability).ToArray();
+
+            return GenerateRankResponse(originalActions, rankableActions, excludedActions, rankedIndices, rankingProbabilities, eventId);
+        }
+
+        private static PersonalizerRankResult GenerateRankResponse(List<PersonalizerRankableAction> originalActions,
```

Comment:
`GenerateRankResponse` and `GenerateRankResult` should have the same name. It's confusing otherwise. Also, I don't see the benefit of the two methods. Why can't all the logic be done in `GenerateRankeResult`?

Reply:
I changed the GenerateRankResponse to GenerateRankResultInner. I don't want to combine the 2 methods as my next PR for multi-slot will call GenerateRankResultInner with different parameters.